src/node.ts

```diff
@@ -1,33 +1,340 @@
-import { Decl } from "./declaration";
-import { Err } from "./error";
-import { Expr } from "./expression";
-import { Stmt } from "./statement";
+import type { Decl, ParameterDecl } from "./declaration";
+import type { Err } from "./error";
+import type { Expr } from "./expression";
```

Comment:
I didn't know you could do this! Strange semantics through, why isn't is `import declaration`?

Reply:
Not sure what your question is, but it's super useful for avoiding module circular references.